### PR TITLE
fix/TagUrl to TS

### LINKED ISSUE #19126: Migrate components to TS: TagUrl
### Description
Migrate `TagUrl` component to TypeScript

### Technical Details

- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` (name and keys)
- Ensure enums use singular not plural e.g. `Size` not `Sizes`
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation

### Acceptance Criteria

- Component has been converted to TypeScript
- Stories, documentation and tests have been updated inside of the `component-name` folder
- Before/After screen cast of component stories in storybook to show no visual regression after migration

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with TypeScript and want to contribute to improving the stability of the extension

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation

- Fixes #19126 

### Before

https://github.com/MetaMask/metamask-extension/assets/79097544/324a8003-e369-4ebd-8bac-6b09e62a69e1

### After

https://github.com/MetaMask/metamask-extension/assets/79097544/bb7b2f18-829b-4a23-8ca3-223deec9d698

unit tests
image

## Manual Testing Steps

- Check storybook controls for `TagUrl` on this branch
- `yarn jest ui/components/component-library/tag-url`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, could you please fix the type errors and failing snapshot tests and I will review. Let me know if you need any support. Thanks!

File: ui/components/component-library/tag-url/tag-url.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        paddingLeft={2}
+        paddingRight={4}
+        gap={2}
+        borderRadius={BorderRadius.pill}
+        display={Display.Flex}
+        {...(props as BoxProps<C>)}
+      >
+        <AvatarFavicon src={src} name={label} {...avatarFaviconProps}>
+          {children}
+        </AvatarFavicon>
```
Comment (dhruvv173) [inline ui/components/component-library/tag-url/tag-url.tsx:59]: the TS error was due to not passing `children` here. 
@georgewrmarshall  Could you please check this when you do a full review and confirm if it is acceptable? Thank you!

Comment (georgewrmarshall) [reply ui/components/component-library/tag-url/tag-url.tsx]: Hey @dhruvv173, I think the move here is to omit it from `AvatarFavicon` as a prop. I've made that update everything else looks pretty good 💯

File: ui/components/component-library/avatar-favicon/avatar-favicon.test.tsx [hunk 2]
```diff
@@ -99,7 +99,6 @@ describe('AvatarFavicon', () => {
     const { getByTestId } = render(
       <AvatarFavicon
         className="mm-avatar-favicon--test"
-        name="test"
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-favicon/avatar-favicon.test.tsx:102]: Was getting a duplicate prop type error in the test file

File: ui/components/component-library/avatar-favicon/avatar-favicon.types.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ export enum AvatarFaviconSize {
 }
 
 export interface AvatarFaviconStyleUtilityProps
-  extends Omit<AvatarBaseStyleUtilityProps, 'size'> {
+  extends Omit<AvatarBaseStyleUtilityProps, 'size' | 'children'> {
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-favicon/avatar-favicon.types.ts:15]: Omitting children as we don't use allow `children` prop to be used in `AvatarFavicon`

File: ui/components/component-library/tag-url/README.mdx [hunk 4]
```diff
@@ -120,9 +130,9 @@ import { TagUrl } from '../../ui/component-library/tag-url';
...
 />
 <TagUrl
-  src="https://metamask.io/icons/icon-48x48.png?v=48400a28770e10dd52a8c0e539aeb282"
+  src="https://metamask.github.io/test-dapp/metamask-fox.svg"
   showLockIcon
-  label="metamask.io"
+  label="metamask.github.io"
 />
 <TagUrl
   src=""
```
Comment (georgewrmarshall) [inline ui/components/component-library/tag-url/README.mdx:138]: Made some small updates to the docs page to show missing images and improve the sentence structure of some sections

File: ui/components/component-library/tag-url/tag-url.types.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+   */
+  labelProps?: TextProps<'p'>;
+  /**
+   * If we want a button in TagUrl component.
+   */
+  actionButtonLabel?: string | React.ReactNode;
+  /**
+   * It accepts all the props from ButtonLink
+   */
+  actionButtonProps?: ButtonLinkProps<'button'>;
```
Comment (georgewrmarshall) [inline ui/components/component-library/tag-url/tag-url.types.ts:43]: Hmm I'm not sure how this component will act if we pass it an href or attributes related to an `a` tag. I've created some stories that use href and they seem to be working without issue so we can monitor it.

**Review by georgewrmarshall — APPROVED** [review]
LGTM! I've made some small updates during my review. Otherwise excellent work! Thanks for your contribution @dhruvv173 💯

**Review by garrettbear — APPROVED** [review]
Nice

**Pull request merged**